tracer/src/Datadog.Trace/Sampling/DefaultSamplingRule.cs

```diff
@@ -58,23 +61,34 @@ public float GetSamplingRate(Span span)
                 return sampleRate;
             }
 
-            Log.Debug("Could not establish sample rate for trace {TraceId}", span.TraceId);
+            Log.Debug("Could not establish sample rate for trace {TraceId}. Using default rate instead: {rate}", span.TraceId, _defaultSamplingRate);
 
-            return 1;
+            return _defaultSamplingRate;
         }
 
         public void SetDefaultSampleRates(IEnumerable<KeyValuePair<string, float>> sampleRates)
         {
             // to avoid locking if writers and readers can access the dictionary at the same time,
             // build the new dictionary first, then replace the old one
             var rates = new Dictionary<SampleRateKey, float>();
+            float defaultSamplingRate = 1;
 
             if (sampleRates != null)
             {
+                // Find the default rate first
+                foreach (var pair in sampleRates)
+                {
+                    if (string.Equals(pair.Key, DefaultKey, StringComparison.OrdinalIgnoreCase))
+                    {
+                        defaultSamplingRate = pair.Value;
+                        break;
+                    }
+                }
+
                 foreach (var pair in sampleRates)
                 {
                     // No point in adding default rates
```

Comment:
If we don't add the default rate, how can we distinguish between "we're using the default rate we got from the agent" and "we're using the default default rate because we don't have any rates from the agent yet"?

Reply:
Indeed we can't. 
I removed altogether this optimization as at least 2 other languages aren't doing that and as it can make investigations harder. 
This leads to a bit more allocations when we receive the rates (and a bit more allocations on every spans as we now set the tag on every span)